From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vincent Strubel <clipos@ssi.gouv.fr>
Date: Wed, 3 Aug 2011 17:16:37 +0200
Subject: [PATCH 20/32] clip: fs: Fix 'readonly' bind mounts

Fix 'readonly' bind mounts that got dropped from vserver patch somewhere in
between 2.6.38 and 2.6.39.

Patch: 1520_bind_mount_readonly.patch
---
 fs/namespace.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/namespace.c b/fs/namespace.c
index 261464dc4aad9..d6f9708991740 100644
--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -2247,6 +2247,7 @@ static int do_loopback(struct path *path, const char *old_name,
 		umount_tree(mnt, UMOUNT_SYNC);
 		unlock_mount_hash();
 	}
+	mnt->mnt.mnt_flags = mnt_flags;
 out2:
 	unlock_mount(mp);
 out:
-- 
2.17.0

